OSDOCS-2197 STS: delete OIDC keys post-installation (#32722)
openshift/openshift-docs: modules/cco-ccoctl-deleting-sts-resources.adoc

// Module included in the following assemblies:
//
// * installing/installing_aws/uninstalling-cluster-aws.adoc

[id="cco-ccoctl-deleting-sts-resources_{context}"]

> = Deleting AWS resources with the Cloud Credential Operator utility

Reviewer: Do you need "utility" at the end? I'm guessing so but I don't recall seeing this for other Operators, so just asking.

Author: yeah, the utility is a separate thing - thanks :)

To clean up resources after uninstalling an {product-title} cluster with the Cloud Credential Operator (CCO) in manual mode with STS, you can use the CCO utility (`ccoctl`) to remove the AWS resources that `ccoctl` created during installation.

.Prerequisites

. Extract and prepare the `ccoctl` binary.

> . Install an {product-title} cluster with the CCO in manual mode with STS.

Reviewer: Should these be bullets instead of numbers?

Author: Hm, I suppose they do not need to be done int that order now that I think about it - good eye!

// Module included in the following assemblies:
//
// * post_installation_configuration/cluster-tasks.adoc

[id="cco-ccoctl-deleting-sts-resources_{context}"]
= Deleting AWS resources with the Cloud Credential Operator utility

> After installing an {product-title} cluster with the Cloud Credential Operator (CCO) in manual mode with STS, you can use the CCO utility (`ccoctl`) to remove the AWS resources that `ccoctl` created. These resources are required only during changes that require elevated permissions, such as upgrades.

Reviewer 2: ```suggestion
After installing an {product-title} cluster with the Cloud Credential Operator (CCO) in manual mode with STS, you can use the CCO utility (`ccoctl`) to remove the AWS resources that `ccoctl` created.
```

All the resources created by ccoctl are necessary while the cluster is running. Some of those resources may need updating during upgrades, but the resources are pretty much always needed.

> This procedure can also be used to clean up resources after uninstalling a cluster that is configured to use the CCO in manual mode with STS.

Reviewer 2: ```suggestion
This procedure is meant to be used to clean up resources previously created by ccoctl after uninstalling a cluster that is configured to use the CCO in manual mode with STS.
```
Some background on how this works. If some of the resources ccoctl can create were created manually in a way that the names were customized before or after creation (ccoctl allows you to mix and match steps of desired/required) the tool will not be able to locate and delete the resources that don't follow ccoctl's naming/tagging scheme. These resources would need to be located and cleaned up manually. To;Dr the delete support assumes ccoctl created everything.

Author: Ok, cool - thanks for that :) I think I was a bit caught up between local and AWS-side items, and then also making this sort of in-line with other CCO topics, but this works inherently differently in that respect :+1:

.Prerequisites

. Extract and prepare the `ccoctl` binary.
. Install an {product-title} cluster with the CCO in manual mode with STS.

.Procedure

* Delete the AWS resources that `ccoctl` created:
+
[source,terminal,subs="+quotes"]
----
$ ccoctl aws delete --name=__<name>__ --region=__<aws_region>__
----
+
where:
+

> ** `_<name>_` is the name used to tag and track the cloud resources that will be deleted.

Reviewer 2: ```suggestion
** `_<name>_` matches the name used to originally create and tag the cloud resources.
```
